sermant-plugins/sermant-springboot-registry/springboot-registry-plugin/src/main/java/io/sermant/discovery/config/NacosRegisterConfig.java

```diff
@@ -0,0 +1,402 @@
+        this.failureToleranceEnabled = failureToleranceEnabled;
+    }
+
+    public int getPaginationSize() {
+        return paginationSize;
+    }
+
+    public void setPaginationSize(int paginationSize) {
+        this.paginationSize = paginationSize;
+    }
+
+    public long getLookupInterval() {
+        return lookupInterval;
+    }
+
+    public void setLookupInterval(long lookupInterval) {
+        this.lookupInterval = lookupInterval;
+    }
+
+    public long getNotifyDelay() {
+        return notifyDelay;
+    }
+
+    public void setNotifyDelay(long notifyDelay) {
+        this.notifyDelay = notifyDelay;
+    }
+
+    public String getPrivateKey() {
+        return privateKey;
+    }
+
+    public void setPrivateKey(String privateKey) {
+        this.privateKey = privateKey;
+    }
+
+    public boolean isEnableAuth() {
+        return enableAuth;
+    }
+
+    public void setEnableAuth(boolean enableAuth) {
+        this.enableAuth = enableAuth;
+    }
+
+    /**
+     * Get configuration parameters
+     *
+     * @return Properties
+     */
+    public Properties getNacosProperties() {
+        LbConfig lbConfig = PluginConfigManager.getPluginConfig(LbConfig.class);
+        Properties properties = new Properties();
+        if (enableAuth) {
+            if (StringUtils.isEmpty(username) || StringUtils.isEmpty(password)
+                    || StringUtils.isEmpty(privateKey)) {
+                LOGGER.log(Level.SEVERE, "Nacos username, password or privateKey is Empty");
+                return properties;
+            }
+            Optional<String> pwd = AesUtil.decrypt(privateKey, password);
+            if (!pwd.isPresent()) {
+                LOGGER.log(Level.SEVERE, "Nacos password parsing failed");
+                return properties;
+            }
+            properties.put(PropertyKeyConst.USERNAME, username);
+            properties.put(PropertyKeyConst.PASSWORD, pwd.get());
+        }
+        properties.put(PropertyKeyConst.SERVER_ADDR, lbConfig.getRegistryAddress());
+        properties.put(PropertyKeyConst.NAMESPACE, Objects.toString(namespace, ""));
+        properties.put(PropertyKeyConst.NACOS_NAMING_LOG_NAME, Objects.toString(logName, ""));
+        if (endpoint.contains(PropertyKeyConst.HTTP_URL_COLON)) {
+            int index = endpoint.indexOf(PropertyKeyConst.HTTP_URL_COLON);
+            properties.put(PropertyKeyConst.ENDPOINT, endpoint.substring(0, index));
+            properties.put(PropertyKeyConst.ENDPOINT_PORT, endpoint.substring(index + 1));
+        } else {
+            properties.put(PropertyKeyConst.ENDPOINT, endpoint);
+        }
+        properties.put(PropertyKeyConst.ACCESS_KEY, Objects.toString(accessKey, ""));
+        properties.put(PropertyKeyConst.SECRET_KEY, Objects.toString(secretKey, ""));
+        properties.put(PropertyKeyConst.CLUSTER_NAME, clusterName);
+        properties.put(PropertyKeyConst.NAMING_LOAD_CACHE_AT_START, namingLoadCacheAtStart);
+        return properties;
```

Comment:
endpoint is never set and default value is "", check is it necessary to config this.  And same as other config keys